x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/assignment/TrainedModelAssignmentClusterService.java

```diff
@@ -537,7 +537,7 @@ Optional<String> handleNodeLoad(NodeLoad load, String nodeId, StartTrainedModelD
                 )
             );
         }
-        if (load.getFreeMemory() < params.estimateMemoryUsageBytes()) {
```

Comment:
This is a bug fix which I can pull out into a separate PR and merge to 8.2 as well.  The fix would look slightly different in 8.2, but never considering the per-node native code overhead here is a bug in released product.